cmd/node/node.go

```diff
@@ -366,7 +366,22 @@ func (app *SpacemeshApp) initServices(nodeID types.NodeId, swarm service.Service
 		hOracle = eligibility.New(beacon, atxdb, BLS381.Verify2, vrfSigner, uint16(app.Config.LayersPerEpoch), lg.WithName("hareOracle"))
 	}
 
-	ha := hare.New(app.Config.HARE, swarm, sgn, nodeID, syncer.IsSynced, msh, hOracle, uint16(app.Config.LayersPerEpoch), idStore, atxdb, clock.Subscribe(), lg.WithName("hare"))
+	// a function to validate we know the blocks
+	validationFunc := func(ids []types.BlockID) bool {
```

Comment:
you dont have do define it here,
 hare knows mesh mesh has access to the GetBLOCK method why not use it ?

Reply:
just change orphanBlockProvider and add GetBLock to it also rename it to BlockProvider or   something

Reply:
Yes, but the Hare doesn't know the mesh as a mesh. We receive the mesh as a block provider...